components/user_profile/user_profile.tsx

```diff
@@ -61,13 +69,15 @@ export default class UserProfile extends PureComponent {
             userId,
         } = this.props;
 
-        let name;
-        if (displayUsername) {
+        let name: React.ReactNode;
+        if (user && displayUsername) {
```

Comment:
To double check, is this a change in behaviour from before? I guess we probably only passed a user ID before when `displayUsername` was true.

Reply:
The problem is on the line below. We can't do `user.username` if `user` is `null` or `undefined`.  I think this might have been a potential source of runtime crash and the type checker highlights it.

I am not 100% sure this is actually changing the behaviour in any meaningful way. It might be good to have a more in-depth look from someone that has a bit more expertise on actually "using" the component in different contexts.

I did check all the available snapshots and this didn't seem to break any tested behavior.

Reply:
I don't think `user` should be undefined here, it should be a required prop. We can define that above and avoid this check.

Reply:
@devinbinnie Unfortunately, I think `user` can be undefined when this is used for system messages, so it can't be required.

@lmammino I think it's probably not a functional change looking more into this. If I had to guess, `props.userId` and `props.displayUsername` are always used together currently even though there's no way to represent that in the type definition. This is good to leave as-is then.